--- a/app/Fact.php	2024-11-30 16:11:46.000000000 +0100
+++ b/app/Fact.php	2025-07-20 04:56:28.143757561 +0200
@@ -444,10 +444,17 @@
      * @return string
      */
     public function label(): string
-    {
-        if (str_ends_with($this->tag(), ':NOTE') && preg_match('/^@' . Gedcom::REGEX_XREF . '@$/', $this->value())) {
-            return I18N::translate('Shared note');
+    {   //+++ =-Fix-P001-=
+        if (str_ends_with($this->tag(), ':NOTE') && preg_match('/^@(' . Gedcom::REGEX_XREF . ')@$/', $this->value(), $match)) {
+            $note = Registry::noteFactory()->make($match[1], $this->record->tree());
+
+            if ($note === null || !$note->canShow()) {
+                return I18N::translate('Shared note');
+            } else {
+                return '<a href="' . e($note->url()) . '">' . I18N::translate('Shared note') . '</a>';
+            }
         }
+        //--- =-Fix-P001-=
 
         // Marriages
         if ($this->tag() === 'FAM:MARR') {
