--- a/resources/views/layouts/administration.phtml	2024-11-30 16:11:46.000000000 +0100
+++ a/resources/views/layouts/administration.phtml	2025-07-20 04:56:06.195735242 +0200
@@ -38,8 +38,8 @@
         <?= View::stack('styles') ?>
 
         <?= Registry::container()->get(ModuleService::class)->findByInterface(ModuleGlobalInterface::class)->map(static function (ModuleGlobalInterface $module): string {
-    return $module instanceof ModuleCustomInterface || $module instanceof CustomCssJsModule ? '' : $module->headContent();
-        })->implode('') ?>
+    return $module instanceof ModuleCustomInterface && !preg_match('/LinkEnhancerModule$/i', get_class($module)) || $module instanceof CustomCssJsModule ? '' : $module->headContent();
+        })->implode('') /* =-Fix-P002-= */?>
     </head>
     <body class="container-lg wt-global wt-control-panel">
         <header>
@@ -89,7 +89,7 @@
         <?= View::stack('javascript') ?>
 
         <?= Registry::container()->get(ModuleService::class)->findByInterface(ModuleGlobalInterface::class)->map(static function (ModuleGlobalInterface $module): string {
-    return $module instanceof ModuleCustomInterface || $module instanceof CustomCssJsModule ? '' : $module->bodyContent();
-        })->implode('') ?>
+    return $module instanceof ModuleCustomInterface && !preg_match('/LinkEnhancerModule$/i', get_class($module)) || $module instanceof CustomCssJsModule ? '' : $module->bodyContent();
+        })->implode('') /* =-Fix-P002-= */ ?>
     </body>
 </html>
