diff -urwB a/GedcomEditService.php b/GedcomEditService.php
--- a/app/Services/GedcomEditService.php	2025-07-20 23:00:52.385936498 +0200
+++ b/app/Services/GedcomEditService.php	2025-07-20 22:40:23.055078084 +0200
@@ -263,6 +263,7 @@
         $next_level = substr_count($tag, ':') + 1;
         $factory    = Registry::elementFactory();
         $subtags    = $factory->make($tag)->subtags();
+        $hasUid     = str_contains($gedcom, "\n1 _UID "); //--- =-Fix-P003-=
 
         // The first part is level N.  The remainder are level N+1.
         $parts  = preg_split('/\n(?=' . $next_level . ')/', $gedcom);
@@ -297,7 +298,7 @@
             }
 
             // Allowed to have more of this subtag?
-            if ($count < $max) {
+            if ($count < $max && (($subtag != '_UID') || ($subtag == '_UID' && !$hasUid))) { //--- =-Fix-P003-=
                 // Create a new one.
                 $gedcom  = $next_level . ' ' . $subtag;
                 $default = $factory->make($tag . ':' . $subtag)->default($tree);
